### ISSUE TITLE
NextJS form post parsing issue

### ISSUE THREAD

**Issue by rishabhpoddar** [supertokens/supertokens-node#496]
The form post parsing throws an error and causes sign in with apple to fail for nextjs apps: https://github.com/supertokens/supertokens-node/blob/f224a42aa1c2cc1f9e008ff87a7adf2fd0b5c29c/lib/ts/framework/utils.ts#L217

Also, the error message that's thrown from the function is not correct + the comments are not correct either.

If we comment out the entire function (pointed above), then it works as expected - but then if someone has disabled the default parser, then how would it work?

**Comment by rishabhpoddar** [comment]
The way that this can be replicated is to:
- start a nextjs app using the create-supertokens-app CLI
- Send the following curl command:
```bash
curl --location --request POST 'http://localhost:3000/api/auth/callback/apple' \
--header 'authority: qa.dvcforless.com' \
--header 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7' \
--header 'accept-language: en-GB,en-US;q=0.9,en;q=0.8' \
--header 'cache-control: no-cache' \
--header 'content-type: application/x-www-form-urlencoded' \
--header 'cookie: _ga_X7SFNZRRT2=GS1.1.1677391130.1.1.1677391366.0.0.0' \
--header 'origin: https://appleid.apple.com' \
--header 'pragma: no-cache' \
--header 'referer: https://appleid.apple.com/' \
--header 'sec-ch-ua: "Chromium";v="110", "Not A(Brand";v="24", "Google Chrome";v="110"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'sec-ch-ua-platform: "macOS"' \
--header 'sec-fetch-dest: document' \
--header 'sec-fetch-mode: navigate' \
--header 'sec-fetch-site: cross-site' \
--header 'sec-fetch-user: ?1' \
--header 'upgrade-insecure-requests: 1' \
--header 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36' \
--data-urlencode 'state=afc596274293e1587315c' \
--data-urlencode 'code=c7685e261f98e4b3b94e34b3a69ff9cf4.0.rvxt.eE8rO__6hGoqaX1B7ODPmA'
```

**Comment by rishabhpoddar** [comment]
This issue has been fixed in node SDK version >= 13.1.2
